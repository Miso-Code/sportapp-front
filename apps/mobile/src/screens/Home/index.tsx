import React from 'react'

import TestComponent from 'components/TestComponent'
import { SafeAreaView } from 'react-native'
import {
	ActivityIndicator,
	Button,
	Icon,
	MD2Colors,
	MD3Colors,
	PaperProvider
} from 'react-native-paper'
import { Colors } from 'react-native/Libraries/NewAppScreen'

function App(): React.JSX.Element {
	const backgroundStyle = {
		backgroundColor: Colors.lighter
	}

	return (
		<PaperProvider>
			<SafeAreaView style={backgroundStyle}>
				<ActivityIndicator animating={true} color={MD2Colors.red800} />
				<TestComponent />
				<Button
					icon='camera'
					mode='contained'
					onPress={() => console.log('Pressed')}>
					Press me
				</Button>
				<Icon source='camera' color={MD3Colors.error50} size={20} />
			</SafeAreaView>
		</PaperProvider>
	)
}

export default App
