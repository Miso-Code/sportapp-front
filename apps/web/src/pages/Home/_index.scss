.home {
	@apply bg-white;

	&-section {
		@apply mt-10;
	}
}
