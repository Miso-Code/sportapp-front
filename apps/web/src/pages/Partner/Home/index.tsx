import NavbarTop from '@/components/NavbarTop'
import Spinner from '@/components/Spinner'
import TransitionAlert from '@/components/TransitionAlert'
import {
	Avatar,
	Box,
	Card,
	CardContent,
	Container,
	Switch,
	Typography
} from '@mui/material'
import { Product } from '@sportapp/sportapp-repository/src/business-partner/interfaces/api/product'

import { usePartnerProductStore } from '@sportapp/stores'
import { useCallback, useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'

export default function HomePartner() {
	const { t } = useTranslation()
	const { getProducts } = usePartnerProductStore()
	const { loading, error } = usePartnerProductStore()
	const [alert, setAlert] = useState(false)
	const [products, setProducts] = useState<Product[]>([])

	const handleGetProducts = useCallback(async () => {
		const response = await getProducts({ limit: 5, offset: 0 })

		if (response) setProducts(response)
	}, [getProducts])

	useEffect(() => {
		window.scrollTo(0, 0)
	}, [])

	useEffect(() => {
		handleGetProducts()
	}, [handleGetProducts])

	return (
		<>
			<div className='home-partner-page'>
				<NavbarTop />
				<Container maxWidth='xl' className='mt-4'>
					<Typography className='mb-4' variant='h3'>
						{t('productObtain.title')}
					</Typography>

					{loading ? (
						<Box
							sx={{
								display: 'grid',
								placeContent: 'center',
								width: '100%',
								height: '100%'
							}}>
							<Spinner />
							<Typography className='my-4' variant='h6'>
								{t('loading')}
							</Typography>
						</Box>
					) : (
						<Box
							style={{
								display: 'grid',
								gridTemplateColumns:
									'repeat(auto-fill, minmax(300px, 1fr))',
								gap: '2rem',
								marginTop: '2rem',
								paddingBottom: '2rem'
							}}>
							{products.map((product) => (
								<Card key={product.product_id}>
									<CardContent>
										<div className='flex justify-between'>
											<div className='flex'>
												<Avatar variant='square'>
													{product.name[0]}
												</Avatar>
												<Typography
													className='ml-4'
													variant='h6'>
													{t(product.name)}
												</Typography>
											</div>
											<Switch
												defaultChecked={product.active}
											/>
										</div>
									</CardContent>
									<CardContent>
										<Typography
											style={{
												height: '4rem',
												overflow: 'hidden',
												textOverflow: 'ellipsis',
												display: '-webkit-box',
												WebkitLineClamp: 3,
												WebkitBoxOrient: 'vertical'
											}}
											variant='caption'>
											{product.summary}
										</Typography>
									</CardContent>
								</Card>
							))}
						</Box>
					)}
				</Container>
			</div>
			<TransitionAlert
				containerClassName='alert-partner-login-container'
				isOpen={alert}
				handleClose={setAlert}
				message={t(error ?? 'errors.partner.createProduct.base')}
				severity='error'
			/>
		</>
	)
}
