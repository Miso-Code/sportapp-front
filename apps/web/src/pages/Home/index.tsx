import { useTranslation } from 'react-i18next'
import './_index.scss'
import Button from '@mui/material/Button'

function Home() {
	const { t } = useTranslation()

	return (
		<>
			<h1 className='text-9xl'>{t('app.name')}</h1>
			<Button variant='contained'>Hello world</Button>
		</>
	)
}

export default Home
