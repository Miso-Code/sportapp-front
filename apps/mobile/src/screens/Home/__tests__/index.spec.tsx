import React from 'react'
import Home from '@/screens/Home'

import renderer, { ReactTestRenderer } from 'react-test-renderer'

describe('Home', () => {
	let wrapper: ReactTestRenderer

	beforeEach(() => {
		wrapper = renderer.create(<Home />)
	})

	afterEach(() => {
		wrapper.unmount()
	})

	it('should render correctly', () => {
		expect(wrapper).toMatchSnapshot()
	})
})
