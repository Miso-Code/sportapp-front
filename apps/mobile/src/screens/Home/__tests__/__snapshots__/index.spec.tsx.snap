// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <RCTSafeAreaView
      style={
        {
          "backgroundColor": "#F3F3F3",
        }
      }
    >
      <View
        accessibilityRole="progressbar"
        accessibilityState={
          {
            "busy": true,
          }
        }
        accessible={true}
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          collapsable={false}
          style={
            {
              "height": 24,
              "opacity": 1,
              "width": 24,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "height": 24,
                  "transform": [
                    {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 12,
                    "overflow": "hidden",
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "height": 24,
                      "transform": [
                        {
                          "translateY": 0,
                        },
                        {
                          "rotate": "-165deg",
                        },
                      ],
                      "width": 24,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "height": 12,
                        "overflow": "hidden",
                        "width": 24,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "borderColor": "#c62828",
                          "borderRadius": 12,
                          "borderWidth": 2.4,
                          "height": 24,
                          "width": 24,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "height": 24,
                  "transform": [
                    {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 12,
                    "overflow": "hidden",
                    "top": 12,
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "height": 24,
                      "transform": [
                        {
                          "translateY": -12,
                        },
                        {
                          "rotate": "345deg",
                        },
                      ],
                      "width": 24,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "height": 12,
                        "overflow": "hidden",
                        "width": 24,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "borderColor": "#c62828",
                          "borderRadius": 12,
                          "borderWidth": 2.4,
                          "height": 24,
                          "width": 24,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Text>
        app.name
      </Text>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(103, 80, 164, 1)",
            "borderRadius": 20,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(103, 80, 164, 1)",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flex": undefined,
              "minWidth": 64,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="button-container"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 20,
                },
              ]
            }
            testID="button"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "marginLeft": 12,
                      "marginRight": -4,
                    },
                    {
                      "marginLeft": 16,
                      "marginRight": -16,
                    },
                    false,
                  ]
                }
                testID="button-icon-container"
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "color": "rgba(255, 255, 255, 1)",
                        "fontSize": 18,
                      },
                      [
                        {
                          "lineHeight": 18,
                          "transform": [
                            {
                              "scaleX": 1,
                            },
                          ],
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                      ],
                      {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  ó°„€
                </Text>
              </View>
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        false,
                        {
                          "marginHorizontal": 24,
                          "marginVertical": 10,
                        },
                        undefined,
                        false,
                        {
                          "color": "rgba(255, 255, 255, 1)",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        undefined,
                      ],
                    ],
                  ]
                }
                testID="button-text"
              >
                Press me
              </Text>
            </View>
          </View>
        </View>
      </View>
      <Text
        accessibilityElementsHidden={true}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          [
            {
              "color": "rgba(220, 54, 46, 1)",
              "fontSize": 20,
            },
            [
              {
                "lineHeight": 20,
                "transform": [
                  {
                    "scaleX": 1,
                  },
                ],
              },
              {
                "backgroundColor": "transparent",
              },
            ],
            {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ó°„€
      </Text>
    </RCTSafeAreaView>
  </View>
</RNCSafeAreaProvider>
`;
